zng/resolver/context.go

```diff
@@ -202,17 +202,36 @@ func (c *Context) AddType(t zng.Type) zng.Type {
 // and returned.  The closure of types within the columns must all be from
 // this type context.  If you want to use columns from a different type context,
 // use TranslateTypeRecord.
-func (c *Context) LookupTypeRecord(columns []zng.Column) *zng.TypeRecord {
+func (c *Context) LookupTypeRecord(columns []zng.Column) (*zng.TypeRecord, error) {
+	// First check for duplicate columns
+	names := make(map[string]struct{})
+	var val struct{}
+	for _, col := range columns {
+		_, exists := names[col.Name]
+		if exists {
+			return nil, fmt.Errorf("duplicate column %s", col.Name)
```

Comment:
if the terminology is column, could you also update the duplicate fields error message in proc/colbuilder.go to refer to column rather than fields?

Reply:
Oops, i think i like field better, i changed it here.  I don't feel strongly though, we can switch to column everywhere if people prefer that.